core/transaction.go

```diff
@@ -409,6 +423,21 @@ func transactionCommitment(transactions []Transaction) (*felt.Felt, error) {
 	})
 }
 
+// ParseBlockVersion computes the block version, defaulting to "0.0.0" for empty strings
+func ParseBlockVersion(protocolVersion string) (*semver.Version, error) {
```

Comment:
This may be a task for a future PR (or totally unnecessary), but we should consider storing the block version as an enum instead of as a string. That way we're not doing regex/string manipulation whenever we calculate a transaction hash.